src/modules/user/user.usecase.ts

```diff
@@ -518,4 +555,36 @@ export class UserUseCases {
       password,
     });
   }
+
+  async updatePassword(
+    user: User,
+    updatePasswordDto: UpdatePasswordDto,
+  ): Promise<void> {
+    const { currentPassword, newPassword, newSalt, mnemonic, privateKey } =
+      updatePasswordDto;
+
+    if (user.password.toString() !== currentPassword) {
```

Comment:
The `toString()` is part of ensuring the format and should be handled by the controller, which ensures the format is correct (in this case, coming from HTTP) before calling the use case.